### ISSUE TITLE
PPE Sign In Error

### ISSUE THREAD

**Issue by cofitz1** [AzureAD/azure-activedirectory-library-for-android#1442]
I'm frequently seeing PPE auth errors with the stacks below

```
2019-07-23T12:55:51.5370000	ERR_	Event 	MSMAM - com.microsoft.intune.mam.client.app.startup.ADALUserAuthentication	      6441	         2	ADAL failed to authenticate
	com.microsoft.aad.adal.AuthenticationException: Error in processing code to get token.  CorrelationId: b32d16d6-3b1d-3c7e-b71f-077c85e9f3a3
		com.microsoft.aad.adal.AcquireTokenInteractiveRequest.acquireTokenWithAuthCode(AcquireTokenInteractiveRequest.java:105)
		com.microsoft.aad.adal.AcquireTokenRequest$3.run(AcquireTokenRequest.java:895)
		java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
		java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
		java.lang.Thread.run(Thread.java:764)
	javax.net.ssl.SSLHandshakeException: Chain validation failed
		com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:355)
		com.android.okhttp.internal.io.RealConnection.connectTls(RealConnection.java:192)
		com.android.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:149)
		com.android.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
		com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
		com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
		com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
		com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
		com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:461)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:127)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:258)
		com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:218)
		com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream(Unknown Source:0)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.setRequestBody(HttpWebRequest.java:251)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.setupConnection(HttpWebRequest.java:127)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.send(HttpWebRequest.java:139)
		com.microsoft.identity.common.adal.internal.net.WebRequestHandler.sendPost(WebRequestHandler.java:74)
		com.microsoft.aad.adal.Oauth2.postMessage(Oauth2.java:653)
		com.microsoft.aad.adal.Oauth2.getTokenForCode(Oauth2.java:632)
		com.microsoft.aad.adal.Oauth2.getToken(Oauth2.java:583)
		com.microsoft.aad.adal.AcquireTokenInteractiveRequest.acquireTokenWithAuthCode(AcquireTokenInteractiveRequest.java:101)
		com.microsoft.aad.adal.AcquireTokenRequest$3.run(AcquireTokenRequest.java:895)
		java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
		java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
		java.lang.Thread.run(Thread.java:764)
	java.security.cert.CertificateException: Chain validation failed
		com.android.org.conscrypt.TrustManagerImpl.verifyChain(TrustManagerImpl.java:705)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:537)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:558)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:626)
		com.android.org.conscrypt.TrustManagerImpl.checkTrusted(TrustManagerImpl.java:493)
		com.android.org.conscrypt.TrustManagerImpl.checkTrusted(TrustManagerImpl.java:416)
		com.android.org.conscrypt.TrustManagerImpl.getTrustedChainForServer(TrustManagerImpl.java:337)
		android.security.net.config.NetworkSecurityTrustManager.checkServerTrusted(NetworkSecurityTrustManager.java:94)
		android.security.net.config.RootTrustManager.checkServerTrusted(RootTrustManager.java:88)
		com.android.org.conscrypt.Platform.checkServerTrusted(Platform.java:203)
		com.android.org.conscrypt.OpenSSLSocketImpl.verifyCertificateChain(OpenSSLSocketImpl.java:592)
		com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
		com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:351)
		com.android.okhttp.internal.io.RealConnection.connectTls(RealConnection.java:192)
		com.android.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:149)
		com.android.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
		com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
		com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
		com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
		com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
		com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:461)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:127)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:258)
		com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:218)
		com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream(Unknown Source:0)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.setRequestBody(HttpWebRequest.java:251)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.setupConnection(HttpWebRequest.java:127)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.send(HttpWebRequest.java:139)
		com.microsoft.identity.common.adal.internal.net.WebRequestHandler.sendPost(WebRequestHandler.java:74)
		com.microsoft.aad.adal.Oauth2.postMessage(Oauth2.java:653)
		com.microsoft.aad.adal.Oauth2.getTokenForCode(Oauth2.java:632)
		com.microsoft.aad.adal.Oauth2.getToken(Oauth2.java:583)
		com.microsoft.aad.adal.AcquireTokenInteractiveRequest.acquireTokenWithAuthCode(AcquireTokenInteractiveRequest.java:101)
		com.microsoft.aad.adal.AcquireTokenRequest$3.run(AcquireTokenRequest.java:895)
		java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
		java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
		java.lang.Thread.run(Thread.java:764)
	java.security.cert.CertPathValidatorException: Response is unreliable: its validity interval is out-of-date
		sun.security.provider.certpath.PKIXMasterCertPathValidator.validate(PKIXMasterCertPathValidator.java:133)
		sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:225)
		sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:143)
		sun.security.provider.certpath.PKIXCertPathValidator.engineValidate(PKIXCertPathValidator.java:79)
		java.security.cert.CertPathValidator.validate(CertPathValidator.java:301)
		com.android.org.conscrypt.TrustManagerImpl.verifyChain(TrustManagerImpl.java:701)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:537)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:558)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:626)
		com.android.org.conscrypt.TrustManagerImpl.checkTrusted(TrustManagerImpl.java:493)
		com.android.org.conscrypt.TrustManagerImpl.checkTrusted(TrustManagerImpl.java:416)
		com.android.org.conscrypt.TrustManagerImpl.getTrustedChainForServer(TrustManagerImpl.java:337)
		android.security.net.config.NetworkSecurityTrustManager.checkServerTrusted(NetworkSecurityTrustManager.java:94)
		android.security.net.config.RootTrustManager.checkServerTrusted(RootTrustManager.java:88)
		com.android.org.conscrypt.Platform.checkServerTrusted(Platform.java:203)
		com.android.org.conscrypt.OpenSSLSocketImpl.verifyCertificateChain(OpenSSLSocketImpl.java:592)
		com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
		com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:351)
		com.android.okhttp.internal.io.RealConnection.connectTls(RealConnection.java:192)
		com.android.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:149)
		com.android.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
		com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
		com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
		com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
		com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
		com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:461)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:127)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:258)
		com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:218)
		com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream(Unknown Source:0)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.setRequestBody(HttpWebRequest.java:251)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.setupConnection(HttpWebRequest.java:127)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.send(HttpWebRequest.java:139)
		com.microsoft.identity.common.adal.internal.net.WebRequestHandler.sendPost(WebRequestHandler.java:74)
		com.microsoft.aad.adal.Oauth2.postMessage(Oauth2.java:653)
		com.microsoft.aad.adal.Oauth2.getTokenForCode(Oauth2.java:632)
		com.microsoft.aad.adal.Oauth2.getToken(Oauth2.java:583)
		com.microsoft.aad.adal.AcquireTokenInteractiveRequest.acquireTokenWithAuthCode(AcquireTokenInteractiveRequest.java:101)
		com.microsoft.aad.adal.AcquireTokenRequest$3.run(AcquireTokenRequest.java:895)
		java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
		java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
		java.lang.Thread.run(Thread.java:764)
	java.security.cert.CertPathValidatorException: Response is unreliable: its validity interval is out-of-date
		sun.security.provider.certpath.OCSPResponse.verify(OCSPResponse.java:619)
		sun.security.provider.certpath.RevocationChecker.checkOCSP(RevocationChecker.java:709)
		sun.security.provider.certpath.RevocationChecker.check(RevocationChecker.java:363)
		sun.security.provider.certpath.RevocationChecker.check(RevocationChecker.java:337)
		sun.security.provider.certpath.PKIXMasterCertPathValidator.validate(PKIXMasterCertPathValidator.java:125)
		sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:225)
		sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:143)
		sun.security.provider.certpath.PKIXCertPathValidator.engineValidate(PKIXCertPathValidator.java:79)
		java.security.cert.CertPathValidator.validate(CertPathValidator.java:301)
		com.android.org.conscrypt.TrustManagerImpl.verifyChain(TrustManagerImpl.java:701)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:537)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:558)
		com.android.org.conscrypt.TrustManagerImpl.checkTrustedRecursive(TrustManagerImpl.java:626)
		com.android.org.conscrypt.TrustManagerImpl.checkTrusted(TrustManagerImpl.java:493)
		com.android.org.conscrypt.TrustManagerImpl.checkTrusted(TrustManagerImpl.java:416)
		com.android.org.conscrypt.TrustManagerImpl.getTrustedChainForServer(TrustManagerImpl.java:337)
		android.security.net.config.NetworkSecurityTrustManager.checkServerTrusted(NetworkSecurityTrustManager.java:94)
		android.security.net.config.RootTrustManager.checkServerTrusted(RootTrustManager.java:88)
		com.android.org.conscrypt.Platform.checkServerTrusted(Platform.java:203)
		com.android.org.conscrypt.OpenSSLSocketImpl.verifyCertificateChain(OpenSSLSocketImpl.java:592)
		com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
		com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:351)
		com.android.okhttp.internal.io.RealConnection.connectTls(RealConnection.java:192)
		com.android.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:149)
		com.android.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
		com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
		com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
		com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
		com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
		com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:461)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:127)
		com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:258)
		com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:218)
		com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream(Unknown Source:0)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.setRequestBody(HttpWebRequest.java:251)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.setupConnection(HttpWebRequest.java:127)
		com.microsoft.identity.common.adal.internal.net.HttpWebRequest.send(HttpWebRequest.java:139)
		com.microsoft.identity.common.adal.internal.net.WebRequestHandler.sendPost(WebRequestHandler.java:74)
		com.microsoft.aad.adal.Oauth2.postMessage(Oauth2.java:653)
		com.microsoft.aad.adal.Oauth2.getTokenForCode(Oauth2.java:632)
		com.microsoft.aad.adal.Oauth2.getToken(Oauth2.java:583)
		com.microsoft.aad.adal.AcquireTokenInteractiveRequest.acquireTokenWithAuthCode(AcquireTokenInteractiveRequest.java:101)
		com.microsoft.aad.adal.AcquireTokenRequest$3.run(AcquireTokenRequest.java:895)
		java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
		java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
		java.lang.Thread.run(Thread.java:764)
```

**Comment by hamiltonha** [comment]
Is this error still occurring?

**Comment by cofitz1** [comment]
It was still occurring up until our recent move to using Prod instead of PPE

**Comment by bannus** [comment]
@hamiltonha @iambmelt We've continued to see this issue after moving our automation from PPE to Prod AAD. The failure pattern is intermittent -- looking back to Jan 2020, it has gone months without happening at all, usually affects 1-3 tests in a run, but on three occasions has occurred dozens of times in a single nightly test run.

We only see these failure on API 26 and above. I've attached a failure stack from an API 26 device below. We have evidence that this can occur on real devices in production as well.

Emulator platform: Ubuntu server
Exception stack: [adal-validity-interval-out-of-date-api-26.log](https://github.com/AzureAD/azure-activedirectory-library-for-android/files/6304959/adal-validity-interval-out-of-date-api-26.log)
Failure screenshot:
<img src="https://user-images.githubusercontent.com/1330640/114566983-ce6caa00-9c40-11eb-89e5-2ea4b76baa51.jpg" alt="failure screenshot" width="200"/>
